### DISCUSSION TITLE
Whats the difference between touched and dirty?

### DISCUSSION THREAD

**Discussion by oschade** [react-hook-form/react-hook-form#5281] [Q&A]
Hello,

the difference between the terms **touched** and **dirty** is not clear to me. I thought, that a field is marked as **dirty** if its content has changed.

I use the sandbox from the documentation example:

https://codesandbox.io/s/react-hook-form-get-started-forked-zpptw?file=/src/index.js

If you place the cursor in a field and leave it (blur) without any changes, the field name appears in touchedFields _and_ dirtyFields. I expected, that it only marked as **touched**.

Is this a bug or expected behaviour?

**Comment by NMinhNguyen** [comment]
~~I'm also curious. Sorry for pinging you directly @bluebill1049 but would you be able to answer this question?~~

Never mind, once you update to the latest version, `dirtyFields` isn't populated unless you actually type into it
